### PR TITLE
[v9, Infra] Fix `FirebaseCore` CMake build error for Firestore CI

### PR DESCRIPTION
**Pull request by ncooke3**
Fixes Firestore CMake CI failues on `v9b`. 

Added a macro to avoid building symbols from new Swift `HeartbeatCoreInternal` module. See comment with context:
https://github.com/firebase/firebase-ios-sdk/blob/614e311d223e3c58bf329258bd86252e78bba8d5/FirebaseCore/CMakeLists.txt#L34-L41

I `ifdef`'d all symbols related to the Swift `FirebaseCoreInternal` module. This is a slightly better bandaid than what I had before.

- I believe this issue surfaced in #9531 when I cherry-picked the `clang-format14` update (which styled a Firestore file).
- This issue is causing Firestore CI failures in #9512.

#no-changelog

### CODE REVIEW

File: FirebaseCore/CMakeLists.txt
Comment (ncooke3) [reply FirebaseCore/CMakeLists.txt]: Thanks Ryan, that looks better. Done in [614e311](https://github.com/firebase/firebase-ios-sdk/pull/9537/commits/614e311d223e3c58bf329258bd86252e78bba8d5).

File: FirebaseCore/Sources/FIRApp.m
Comment (ryanwilson) [inline FirebaseCore/Sources/FIRApp.m:36]: Hmm since everything in the `FIRHeartbeatLogger` header is ifdef'ed out already, is this necessary?

Comment (ncooke3) [reply FirebaseCore/Sources/FIRApp.m:36]: Oh darn, good catch. I did it this way originally. Thanks for thorough review!!

Comment (ncooke3) [reply FirebaseCore/Sources/FIRApp.m]: Done in [42144ea](https://github.com/firebase/firebase-ios-sdk/pull/9537/commits/42144ea844d345a3b5ca3c02591ce66dcb393903)

Comment (ncooke3) [reply FirebaseCore/Sources/FIRApp.m]: I'll wait for CMake CI to go green before merging.

**Comment by ncooke3** [comment]
Some of the SPM CI cancelled due to a failure with the build cache. I'm manually kicking off each of the cancelled workflows.

**Review by schmidt-sebastian — APPROVED** [review]

**Pull request merged**
